tools/integration_tests/util/setup/setup.go

```diff
@@ -262,17 +287,43 @@ func LogAndExit(s string) {
 	os.Exit(1)
 }
 
-// Clean the mounted directory
-func CleanMntDir() {
-	dir, err := os.ReadDir(mntDir)
+// CleanUpDir cleans up the content in given directory.
+func CleanUpDir(directoryPath string) {
+	dir, err := os.ReadDir(directoryPath)
 	if err != nil {
 		log.Printf("Error in reading directory: %v", err)
 	}
 
 	for _, d := range dir {
-		err := os.RemoveAll(path.Join([]string{mntDir, d.Name()}...))
+		err := os.RemoveAll(path.Join([]string{directoryPath, d.Name()}...))
 		if err != nil {
 			log.Printf("Error in removing directory: %v", err)
 		}
 	}
 }
+
+// CleanMntDir cleans the mounted directory.
+func CleanMntDir() {
+	CleanUpDir(mntDir)
+}
+
+// SetupTestDirectory creates a testDirectory in the mounted directory and cleans up
+// any content present in it.
+func SetupTestDirectory(testDirName string) string {
+	testDirPath := path.Join(MntDir(), testDirName)
+	err := os.Mkdir(testDirPath, DirPermission_0755)
+	if err != nil && !strings.Contains(err.Error(), "file exists") {
+		log.Printf("Error while setting up directory %s for testing: %v", testDirPath, err)
+	}
+	CleanUpDir(testDirPath)
```

Comment:
Is this step really necessary. won't os.Mkdir() throw error if directory already exists?

Reply:
Background: @Tulsishah suggested that in future, in order to parallelise the tests, we need to run them in separate directories.

This step is necessary because we are not sure of the order in which the tests are called. I tried creating the test directory before executing the tests as well but we have many different kinds of mount - for eg: dynamic, only-dir, etc where we are not sure of the location where we should create this directory. Even if we do, we still need to update the path, so this change seemed reasonable.

os.Mkdir will throw "file exists" error if directory already exists which we are ignoring here.

Reply:
got it.